commit 84d01ea350c5bd7ade083badf08c49fef46e125b (refs/changes/04/804/2)
Author: Richard Kiene <richard.kiene@joyent.com>
Date:   2016-10-28T03:41:12+00:00 (2 years, 11 months ago)
    
    TOOLS-1590 Write after end errors should be handled in Watershed
    Reviewed by: Josh Wilsdon <jwilsdon@joyent.com>
    Reviewed by: Joshua M. Clulow <jmc@joyent.com>
    Approved by: Joshua M. Clulow <jmc@joyent.com>

diff --git a/CHANGES.md b/CHANGES.md
index 8d7ab59..4de00f6 100644
--- a/CHANGES.md
+++ b/CHANGES.md
@@ -2,6 +2,11 @@
 
 ## not yet released
 
+## 0.3.3
+
+* Handle write-after-end errors when an attempt is made to write to a
+  websocket that cannot be written to.
+
 ## 0.3.2
 
 * Initial move to Joyent org plus make check
diff --git a/lib/watershed.js b/lib/watershed.js
index bab130f..5b16ba4 100644
--- a/lib/watershed.js
+++ b/lib/watershed.js
@@ -326,7 +326,15 @@ WatershedConnection(options, socket)
 			return;
 		self._end_emitted = true;
 
-		if (err.code === 'ECONNRESET' || err.code === 'EPIPE') {
+		/*
+		 * Unfortunately, in the case of a write-after-end error there
+		 * is no error code set. In this case we check that the error
+		 * message property is equal to the string 'write after end' as
+		 * it is specified in the node runtime.
+		 */
+		if (err.code === 'ECONNRESET' ||
+		    err.code === 'EPIPE' ||
+		    err.message === 'write after end') {
 			/*
 			 * Treat end-of-stream errors as merely an end
 			 * of stream.  If we received a CLOSE frame, it
diff --git a/package.json b/package.json
index 0f9a0fb..6de4742 100644
--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
 	"name": "watershed",
-	"version": "0.3.2",
+	"version": "0.3.3",
 	"description": "simple websockets (RFC6455) client and server",
 	"main": "./lib/watershed.js",
 	"author": "Joshua M. Clulow <jmc@joyent.com>",
